packages/grafana-ui/src/components/TimeSeries/utils.ts

```diff
@@ -219,6 +220,58 @@ export const preparePlotConfigBuilder: UPlotConfigPrepFn<{ sync: DashboardCursor
 
   builder.scaleKeys = [xScaleKey, yScaleKey];
 
+  // if hovered value is null, how far we may scan left/right to hover nearest non-null
+  const hoverProximityPx = 10;
+
+  let cursor: Partial<uPlot.Cursor> = {
+    // this scans left and right from cursor position to find nearest data index with value != null
+    // TODO: do we want to only scan past undefined values, but halt at explicit null values?
```

Comment:
the problem with stopping at explicit nulls is that it doesnt fix #34542 since the top chart has a lot of intermittent nulls that are `spanNulls: true`